indexing-service/src/main/java/org/apache/druid/indexing/worker/WorkerTaskManager.java

```diff
@@ -251,7 +251,9 @@ public void onFailure(Throwable t)
           {
             submitNoticeToExec(new StatusNotice(task, TaskStatus.failure(task.getId())));
           }
-        }
+        },
+        // The callback is non-blocking and quick, so it's OK to schedule it using directExecutor()
```

Comment:
Please add a javadoc comment to `submitNoticeToExec()` like "This method is mostly non-blocking because exec is a ThreadPoolExecutor with unbounded queue used by default."

Correspondingly, this comment also should say "mostly non-blocking", not just "non-blocking".